speechbrain/k2_integration/graph_compiler.py

```diff
@@ -1,43 +1,124 @@
 """Graph compiler class to create, store, and use k2 decoding graphs in
-speechbrain. The addition of a decoding graph, be it HL or HLG (with LM),
-limits the output words to the ones in the lexicon. On top of that, a
-bigger LM can be used to rescore the decoding graph and get better results.
+speechbrain. Limits the output words to the ones in the lexicon.
 
 This code is an extension, and therefore heavily inspired or taken from
 icefall's (https://github.com/k2-fsa/icefall) graph compiler.
 
 Authors:
+  * Pierre Champion 2023
   * Zeyu Zhao 2023
   * Georgios Karakasidis 2023
 """
 
 
 import os
 from pathlib import Path
-from typing import List, Union, Optional
+from typing import List, Union, Optional, Tuple
+import abc
 
-try:
-    import k2
-except ImportError:
-    MSG = "Please install k2 to use k2 training \n"
-    MSG += "E.G. run: pip install k2\n"
-    raise ImportError(MSG)
+from . import k2 # import k2 from ./__init__.py
 
 import torch
 import logging
 
-from speechbrain.k2_integration.lexicon import Lexicon
-from speechbrain.k2_integration.utils import (
-    get_texts,
-    one_best_decoding,
-    rescore_with_whole_lattice,
-)
+from . import lexicon, utils
 
 
 logger = logging.getLogger(__name__)
 
+class GraphCompiler(abc.ABC):
```

Comment:
Sorry, might I ask why you created the abstract class `GraphCompiler` first and defined the `CtcGraphCompiler` later on?

Reply:
To later add `LFMMIGraphCompiler` or `RNNTGraphCompiler`!
All of them should define the abstract method/property of `GraphCompiler` for `GraphCompiler.compile` (actually I forgot to this to the abstract class), `GraphCompiler.topo` ..